File: src/game/Bot1.java
package game;

import action.Action;
import ai.Evaluation2;
import ai.MinMax;

public class Bot1 extends Bot {
    public Bot1() {
        super();
        setName("Bot1");
    }
    @Override
    public Action chooseAction() {
        int depth;
        int mobility = mobility(getBoard());

        if (mobility < 9)
            depth = 8;
        else if (mobility < 20)
            depth = 7;
        else
            depth = 6;

        minMax.setEvaluation(new Evaluation2());
        Action action = minMax.decisionAlphaBeta(getBoard(), depth, true);
        System.out.println(getName() + " play " + action);
        return action;
    }
}

File: projectC/awale/src/game/bot1.c
#include "../../headers/game/bot1.h"

Bot1 createBot1()
{
    Bot1 bot1;
    bot1.bot = createBot("Bot1");
    printf("ttt %s\n", bot1.bot.player.name);
    printf("ttt %s\n", bot1.bot.player.name);
    // printf("ttt %d\n", *(bot1.bot->player->nbSeed));
    // bot1.bot->chooseAction = chooseAction_bot1;

    return bot1;
}

Action chooseAction_bot1(Bot *bot)
{
    // int depth = 6;
    // int mob = mobility(bot->player, bot->player->board);

    // if (mob < 9)
    // {
    //     depth = 8;
    // }
    // else if (mob < 20)
    // {
    //     depth = 7;
    // }

    // Evaluation evaluation;
    // evaluation.evaluate = evaluate_evaluation2;
    // setEvaluation(&(bot->minMax), &(evaluation));

    // Action action = decisionAlphaBeta(&(bot->minMax), bot->player->board, depth, true);
    // printf("%s play %s", bot->player->name, toStringAction(&(action)));
    Action action;

    return action;
}
